Test_Task/main.rb

```diff
@@ -0,0 +1,100 @@
+require 'watir'
+require 'nokogiri'
+require 'json'
+require 'date'
+require_relative './transactions'
+
+class WebBanking
+  attr_reader :browser, :page, :operation, :desc_operation
+  def initialize
+    @browser = Watir::Browser.new :chrome
+  end
+
+  def parse_accounts
+    sign_in
+    sleep(3)
+    accounts = browser.divs(class: %w(contract status-active))
+    accounts_info = accounts.map do |div|
+      Watir::Wait.until { div.div(class: "contract-cards").a.present? }
+      div.div(class: "contract-cards").a.click
+      go_to_info
+      go_to_transactions
+      data = {
+          name: parse_name,
+          balance: parse_balance,
+          currency: parse_currency,
+          description: parse_nature,
+          transactions: transactions.map(&:to_hash)
+      }
+      browser.li(class: %w(new_cards_accounts-menu-item active)).a.click
+      data
+    end
+    JSON.pretty_generate(accounts: accounts_info)
+  end
+
+  private
+
+  def sign_in
+    browser.goto 'https://wb.micb.md'
+    browser.text_field(class: "username").set("connicov1")
+    browser.text_field(id: "password").set("Paviliong7")
+    browser.button(class: "wb-button").click
+  end
+
+  def go_to_info
+    browser.ul(class: %w(ui-tabs-nav ui-helper-reset ui-helper-clearfix ui-widget-header ui-corner-all)).lis[1].click
+    @page = Nokogiri::HTML.parse(browser.div(id: "contract-information").html)
+  end
+
+  # I use page as argument just for working with specs
```

Comment:
Are you kidding me?
Do you really modify your code each time you run specs?

🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️ 🤦‍♂️